include/onnxruntime/core/session/onnxruntime_session_options_config_keys.h

```diff
@@ -249,4 +249,10 @@ static const char* const kOrtSessionOptionEpContextFilePath = "ep.context_file_p
 // Flag to specify whether to dump the EP context into the Onnx model.
 // "0": dump the EP context into separate file, keep the file name in the Onnx model.
 // "1": dump the EP context into the Onnx model. (default).
-static const char* const kOrtSessionOptionEpContextEmbedMode = "ep.context_embed_mode";
\ No newline at end of file
+static const char* const kOrtSessionOptionEpContextEmbedMode = "ep.context_embed_mode";
+
+// Enable to dump the EP context node with "hardware_architecture" attribute and check this attribute against the
+// hardware architecture when inferencing.
+// "0": disable. (default)
+// "1": enable.
+static const char* const kOrtSessionOptionEpContextHardwareArchitectureEnable = "ep.context_hardware_architecture_enable";
```

Comment:
is there any downside to always dumping the hardware_architecture attribute (each EP can choose whether or not it is populated with something)  so then we wouldn't need to add yet another option?

Reply:
@gedoensmax
In the case of EP context model pointing/using the TRT engine cache, do you have any comments about compute capability check? What's your deployment env regarding the hardware compatibility? 

TRT provides config (for example: HardwareCompatibilityLevel::kAMPERE_PLUS) to relax compute capability check.
As for TRT EP, we are thinking of removing the options (trt_ep_context_hardware_capability_check),
meaning we always dump Context EP model with "hardware_architecture" attribute included, and when loading the EP context model, we always check the exact match of hardware_architecture/compute_capability.

Reply:
Ok if we give an engine a hard reject based on the HW compatibility this is a problem we certainly want to support cases like Ampere+. 
If we agree that the flag does not signal strict compatibility, but the hardware the engine was built/optimized for we do not need an option for it. Otherwise, yes there are cases where an engine can have multiple hardware compatibilities. We might want to send a warning "Engine was compiled for a different compatibility level and might not work or perform suboptimal" and still "try" to load the engine and then give a final error that is does not work or works.

Reply:
TRT EP won't perform strict hardware compatibility check for EP Context model (This aligns with how TRT EP treats engine cache as well) and will throw a warning message when the HW compute capability mismatches.